Sources/NIO/IO.swift

```diff
@@ -28,16 +32,47 @@ public struct IOError: Swift.Error {
         return .reason(self.failureDescription)
     }
 
+    private enum Error {
+        #if os(Windows)
+            case windows(DWORD)
+            case winsock(CInt)
+        #endif
+        case errno(CInt)
+    }
+
+    private let error: Error
+
     /// The `errno` that was set for the operation.
-    public let errnoCode: CInt
+    public var errnoCode: CInt {
+        switch self.error {
+            case .errno(let code):
+              return code
+            #if os(Windows)
+                default:
+                    fatalError("IOError domain is not `errno`")
+            #endif
```

Comment:
I apologise in advance for this, but...we indent the `case` _declaration_, not the `case` statement in a `switch`. That is:

```swift
enum Foo {
    case bar
    case baz
}

switch aFoo {
case bar:
    break
case baz:
    break
}
```

Do you mind unindenting the `case` statements in the `switch`? (While we're here the `return code` is also under-indented, it should be 4 spaces not 2.